modules/imgcodecs/test/test_png.cpp

```diff
@@ -7,6 +7,21 @@ namespace opencv_test { namespace {
 
 #if defined(HAVE_PNG) || defined(HAVE_SPNG)
 
+TEST(Imgcodecs_Png, imread_passing_mat)
+{
+  const string root = cvtest::TS::ptr()->get_data_path();
+  const string imgName = root + "../cv/shared/lena.png";
+  Mat img = Mat::zeros(1000, 1000, CV_8UC1);
+  Mat subImg = img.rowRange(64, 576).colRange(128, 640);
+  imread(imgName, subImg, IMREAD_GRAYSCALE);
+  Mat ori = imread(imgName, IMREAD_GRAYSCALE);
+  EXPECT_EQ(0, cv::norm(ori, subImg, NORM_INF));
```

Comment:
Get ROI again before the check:

```suggestion
  Mat subImg = img.rowRange(64, 576).colRange(128, 640);
  imread(imgName, subImg, IMREAD_GRAYSCALE);
  
  Mat ori = imread(imgName, IMREAD_GRAYSCALE);
  subImg = img.rowRange(64, 576).colRange(128, 640);
  EXPECT_EQ(0, cv::norm(ori, subImg, NORM_INF));
```